### PR TITLE
gcp - add pubsub initial setup

### PR DESCRIPTION
**Pull request by alexkarpitski**
Added initial classes with `query` and `get` functionality for `Subscription` and `Snapshots` resources in PubSub service

### CODE REVIEW

File: tools/c7n_gcp/c7n_gcp/resources/pubsub.py [hunk 1]
```diff
@@ -34,3 +34,31 @@ class resource_type(TypeInfo):
         def get(client, resource_info):
             return client.execute_command(
                 'get', {'topic': resource_info['topic_id']})
+
+
+@resources.register('pubsub-subscription')
+class PubSubSubscription(QueryResourceManager):
```
Comment (kapilt) [inline tools/c7n_gcp/c7n_gcp/resources/pubsub.py]: uh.. this seems questionable.. whats the use case for a pubsub-subscription resource...

File: tools/c7n_gcp/tests/test_pubsub.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        session_factory = self.replay_flight_data(
+            'pubsub-snapshot-query', project_id=project_id)
+
+        policy = self.load_policy(
+            {'name': 'gcp-pubsub-snapshot-dryrun',
+             'resource': 'gcp.pubsub-snapshot'},
+            session_factory=session_factory)
+
+        pubsub_snapshot_resources = policy.run()
+        self.assertEqual(len(pubsub_snapshot_resources), 1)
```
Comment (kapilt) [inline tools/c7n_gcp/tests/test_pubsub.py]: please assert an attribute of the retrieved resource in the test.

File: tools/c7n_gcp/tests/test_pubsub.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        session_factory = self.replay_flight_data(
+            'pubsub-subscription-query', project_id=project_id)
+
+        policy = self.load_policy(
+            {'name': 'gcp-pubsub-subscription-dryrun',
+             'resource': 'gcp.pubsub-subscription'},
+            session_factory=session_factory)
+
+        pubsub_subscription_resources = policy.run()
+        self.assertEqual(len(pubsub_subscription_resources), 1)
```
Comment (kapilt) [inline tools/c7n_gcp/tests/test_pubsub.py]: please assert an attribute of the retrieved resource in the test

**Comment by kapilt** [comment]
run `make lint` to see and fix flake8 warnings, looks like its mostly wrt to tests

**Comment by alexkarpitski** [comment]
@kapilt, thank you for reviewing, I have fixed the issues.

**Pull request merged**